app/models/post.rb

```diff
@@ -19,9 +19,11 @@ class Post < ApplicationRecord
   after_create :update_posts_counter
 
   def set_defaults
-    if likes.count >= 0 && comments.count >= 0
-      self.likes_counter = likes.count
-      self.comments_counter = comments.count
+    likes_count = Like.includes(:post).where(post: self).references(:post).count
+    comments_count = Comment.includes(:post).where(post: self).references(:post).count
+    if likes_count >= 0 && comments_count >= 0
+      self.likes_counter = likes_count
+      self.comments_counter = comments_count
     else
       self.likes_counter = 0
       self.comments_counter = 0
```

Comment:
- [ ] When I try to delete a `Post` instance that has comments or likes, Ruby on Rails won't permit it if you don't set it to be destroyed with its dependent. I would advise you to set it by using the `dependent: :destroy` clause when defining the association in the model files. 👍 💻

#### This is the error you get when you try to delete a Post instance that has comments or likes:

![image](https://github.com/Movgto/my-blog-app/assets/77106962/32a84696-adfe-4711-a12c-76e6c76288b3)